src/grid_interpolator.cc

```diff
@@ -16,7 +16,7 @@ GridInterpolator::GridInterpolator(const std::string& s)
         }
         
     int nx, ny, nz;   
-    indata >> nx >> ny >> nz >> dx_ >> dy_ >> dz_;
+    indata >> x_ >> cutoff_ >> rho_ >> nx >> ny >> nz >> dx_ >> dy_ >> dz_;
```

Comment:
Yes, here I think you are thinking these are going to be the bounds of your coordinates. You don't need to read them from a file, as you can compute them from the number of points and the spacing along each dimension. If your grid contains the end points (I think it does), then you should do like this:
```suggestion
    indata >> nx >> ny >> nz >> dx_ >> dy_ >> dz_;
    bounds_ = std::make_tuple((nx-1) * dx_, (ny-1) * dy_, (nz-1) * dz_);
```

Reply:
Ohh okay! I will make the changes!